src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1070,6 +1075,19 @@ private UsernamePasswordAuthenticationToken loginAndSetUserData(
             user.setFullName(fullName);
         }
 
+        // Set avatar if possible
+        String avatarUrl = determineStringField(avatarFieldExpr, idToken, userInfo);
+        OicAvatarProperty oicAvatarProperty;
+        if (avatarUrl != null) {
+            LOGGER.finest("Avatar url is: " + avatarUrl);
```

Comment:
this is unconditionally performing string concatenation and in the majority of cases (finest logging is not enabled) is throwing the result away.
Please use a [parameterized](https://docs.oracle.com/en/java/javase/17/docs/api/java.logging/java/util/logging/Logger.html#log(java.util.logging.Level,java.lang.String,java.lang.Object)) [version ](https://docs.oracle.com/en/java/javase/17/docs/api/java.logging/java/util/logging/Logger.html#log(java.util.logging.Level,java.lang.String,java.lang.Object%5B%5D)) (e.g. `LOGGER.log(Level.FINEST, "massage {0} placeholder", new Object[] {"with a"})`  or a [supplier](https://docs.oracle.com/en/java/javase/17/docs/api/java.logging/java/util/logging/Logger.html#fine(java.util.function.Supplier)).